moler/cmd/unix/sftp.py

```diff
@@ -0,0 +1,294 @@
+# -*- coding: utf-8 -*-
+"""
+SFTP command module.
+"""
+__author__ = 'Agnieszka Bylica'
+__copyright__ = 'Copyright (C) 2018, Nokia'
+__email__ = 'agnieszka.bylica@nokia.com'
+
+
+from moler.cmd.unix.genericunix import GenericUnixCommand
+from moler.exceptions import CommandFailure
+from moler.exceptions import ParsingDone
+import re
+
+
+class Sftp(GenericUnixCommand):
+    def __init__(self, connection, host, password, user=None, confirm_connection=True, source_path=None,
+                 destination_path=None, options=None, command=None, no_result=False, prompt=None, new_line_chars=None):
+        super(Sftp, self).__init__(connection=connection, prompt=prompt, new_line_chars=new_line_chars)
+
+        # Attributes defined by calling the command
+        self.host = host
+        self.user = user
+        self.password = password
+        self.confirm_connection = confirm_connection
+
+        self.source_path = source_path
+        self.destination_path = destination_path
+
+        self.options = options
+        self.command = command
+        self.no_result = no_result
+
+        # Internal variables
+        if self.command:
+            self.command = self.command.strip()
+            self._re_command_sent = re.compile(r"sftp> {}".format(re.escape(self.command)), re.I)
+
+        # Flags
+        self.ready_to_parse_line = False
+        self.connection_confirmed = False
+        self.password_sent = False
+        self.sending_started = False
+        self.command_entered = False
+        self.command_sent = False
+        self.exit_sent = False
+
+    def build_command_string(self):
+        cmd = "sftp"
+        if self.options:
+            cmd = "{} {}".format(cmd, self.options)
+        if self.user:
+            cmd = "{} {}@{}".format(cmd, self.user, self.host)
+        else:
+            cmd = "{} {}".format(cmd, self.host)
+        if self.source_path:
+            cmd = "{}:{}".format(cmd, self.source_path)
+        if self.destination_path:
+            cmd = "{} {}".format(cmd, self.destination_path)
+        return cmd
+
+    def on_new_line(self, line, is_full_line):
+        try:
+            if is_full_line:
```

Comment:
Why check twice the same condition? The same condition in line 70.

Reply:
It's because I need to have these methods in this particular order. So I can't check them in one condition.

Reply:
I've fixed this and it works, because I tested it in a terminal. But I had to delete two command tests, because they wouldn't work with only full lines.